src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1093,31 +1047,15 @@ ImmutableArray<FieldSymbol> collectTupleElementFields(NamedTypeSymbol tuple)
                 }
             }
 
-            internal SmallDictionary<FieldSymbol, int> GetFieldDefinitionsToIndexMap(NamedTypeSymbol tuple)
-            {
-                Debug.Assert(tuple.IsTupleType);
-                Debug.Assert(tuple.IsDefinition); // we only store a map for definitions
-                if (_lazyFieldDefinitionsToIndexMap is null)
-                {
-                    tuple.InitializeTupleFieldDefinitionsToIndexMap();
-                }
-
-                Debug.Assert(_lazyFieldDefinitionsToIndexMap is object);
-                return _lazyFieldDefinitionsToIndexMap;
-            }
-
-            internal void SetFieldDefinitionsToIndexMap(SmallDictionary<FieldSymbol, int> map)
-            {
-                Debug.Assert(map.Keys.All(k => k.IsDefinition));
-                Debug.Assert(map.Values.All(v => v >= 0));
-                Interlocked.CompareExchange(ref _lazyFieldDefinitionsToIndexMap, map, null);
-            }
-
             internal SmallDictionary<Symbol, Symbol> UnderlyingDefinitionToMemberMap
             {
                 get
                 {
-                    return _lazyUnderlyingDefinitionToMemberMap ??= computeDefinitionToMemberMap();
+                    if (_lazyUnderlyingDefinitionToMemberMap is null)
+                    {
+                        Interlocked.CompareExchange(ref _lazyUnderlyingDefinitionToMemberMap, computeDefinitionToMemberMap(), null);
```

Comment:
>Interlocked.CompareExchange(ref _lazyUnderlyingDefinitionToMemberMap, computeDefinitionToMemberMap(), null); [](http://example.com/codeflow?start=24&length=108)

It doesn't look like the identity of the dictionary is important. So, it is not clear what difference would this change make.